plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/RewriteExactNumericConstant.java

```diff
@@ -52,7 +55,7 @@ public Optional<ParameterizedExpression> rewrite(Constant constant, Captures cap
             // TODO we could handle NULL values too
             return Optional.empty();
         }
-        if (type == TINYINT || type == SMALLINT || type == INTEGER || type == BIGINT || type instanceof DecimalType) {
+        if (type == TINYINT || type == SMALLINT || type == INTEGER || type == BIGINT || type == REAL || type == DOUBLE || type instanceof DecimalType) {
```

Comment:
The class name -- RewriteExactNumericConstant -- was supposed to mean "not floating point".

You'd need to rename the class if you want to repurpose it.

Reply:
How are NaN values handled? are they uniformly supported by all databases?
Or did RewriteExactNumericConstant become PostgreSQL-specific?

Reply:
But in other places like in ConnectorExpressionRewriter - we do use `numeric_types` for floating point as well right ? 

`.withTypeClass("numeric_type", ImmutableSet.of("tinyint", "smallint", "integer", "bigint", "decimal", "real", "double"))
`
`numeric` means both the floating point and integeral data types right or is it other way round ?

Reply:
> How are NaN values handled? are they uniformly supported by all databases?
Or did RewriteExactNumericConstant become PostgreSQL-specific?

`RewriteExactNumericConstant` exposes the constant as a `QueryParameter` - so when we are building the SQL - `DefaultQueryBuilder` would try to use the `WriteFunction` from the `ColumnMapping` so each database could have its own interpretation of `NaN` and it would be still handled as a part of `ColumnMapping`

Reply:
The real and double types are **_approximate_** numeric type (not **_exact_** numeric) from SQL standard perspective:
```
<numeric type> ::=
    <exact numeric type> 
  | <approximate numeric type> 
  | <decimal floating-point type> 

<exact numeric type> ::=
    NUMERIC [ <left paren>  <precision>  [ <comma>  <scale>  ] <right paren>  ]
  | DECIMAL [ <left paren>  <precision>  [ <comma>  <scale>  ] <right paren>  ]
  | DEC [ <left paren>  <precision>  [ <comma>  <scale>  ] <right paren>  ]
  | SMALLINT
  | INTEGER
  | INT
  | BIGINT

<approximate numeric type> ::=
    FLOAT [ <left paren>  <precision>  <right paren>  ]
  | REAL
  | DOUBLE PRECISION

<decimal floating-point type> ::=
  DECFLOAT [ <left paren>  <precision>  <right paren>  ]
```

Reply:
> But in other places like in ConnectorExpressionRewriter - we do use `numeric_types` for floating point as well right ?
> 
> `.withTypeClass("numeric_type", ImmutableSet.of("tinyint", "smallint", "integer", "bigint", "decimal", "real", "double")) `

That's local to given connector and up to given connector's support for infinity/NaNs and semantical consistency with Trino.
The `RewriteExactNumericConstant` is not connector specific.


> `RewriteExactNumericConstant` exposes the constant as a `QueryParameter` - so when we are building the SQL - `DefaultQueryBuilder` would try to use the `WriteFunction` from the `ColumnMapping` so each database could have its own interpretation of `NaN` and it would be still handled as a part of `ColumnMapping`

Yes. And this may or may not work, as remote database may or may not support +-infinity and NaNs for floating point types (not all do).